### ISSUE TITLE
Error: inferred type is java.lang.CharSequence but kotlin.CharSequence was expected

### ISSUE THREAD

**Issue by C2H6O** [airbnb/epoxy#1043]
👋 Seeing the following error when defining a property of type `CharSequence` in a Kotlin model:

Error: 

```
build/generated/source/kapt/debug/com/plangrid/android/epoxy/models/EpoxyModelKotlinExtensions.kt: (27, 41): Type mismatch: inferred type is java.lang.CharSequence but kotlin.CharSequence was expected
```

Defined model:
```
@EpoxyModelClass(layout = R2.layout.listitem_1_line_selectable_textview)
abstract class SelectableTextModel(
    @EpoxyAttribute(EpoxyAttribute.Option.DoNotHash) var clickListener: (() -> Unit)? = null,
    @EpoxyAttribute var text: CharSequence,
    @EpoxyAttribute var textColor: AlloyTextView.TextColor? = null,
    @EpoxyAttribute var icon: Icon? = null,
    @EpoxyAttribute var selectable: Selectable? = null
) : EpoxyModelWithHolder<SelectableTextModel.Holder>() {
   ...
}
```

Generated model:
```
...
import java.lang.CharSequence
import kotlin.Boolean
import kotlin.Function0
import kotlin.Function1
import kotlin.String
import kotlin.Suppress
import kotlin.Unit
import org.jetbrains.annotations.NotNull
import org.jetbrains.annotations.Nullable

inline fun ModelCollector.selectableText(
    @Nullable noinline clickListener: Function0<Unit>?,
    @NotNull text: CharSequence,
    @Nullable textColor: AlloyTextView.TextColor?,
    @Nullable icon: SelectableTextModel.Icon?,
    @Nullable selectable: Selectable?,
    modelInitializer: SelectableTextModelBuilder.() -> Unit
) {
    add(
    SelectableTextModel_(clickListener, text, textColor, icon, selectable).apply {
        modelInitializer()
    }
    )
}
...
```

I bumped into this when trying make the model more generic by converting the `String` parameter into `CharSequence` so that it can also take `Spanned` (which extends `CharSequence`) as input. 

Cheers.

**Comment by elihart** [comment]
Thanks for reporting, I'll try to get a fix out for this soon

**Comment by elihart** [comment]
I have reproduced it - you can work around it for now by putting the property in the class body instead of in the constructor

**Comment by C2H6O** [comment]
Ah, that's great! Thanks for the workaround

**Comment by elihart** [comment]
Fixed in 4.0 #1046
